portality/api/v1/discovery.py

```diff
@@ -239,59 +230,61 @@ def _make_response(cls, endpoint, res, q, page, page_size, sort,
     def search_articles(cls, q, page, page_size, sort=None):
         search_subs = app.config.get("DISCOVERY_ARTICLE_SEARCH_SUBS", {})
         sort_subs = app.config.get("DISCOVERY_ARTICLE_SORT_SUBS", {})
-        query, page, page_size = cls._make_search_query(q, page, page_size, sort, search_subs, sort_subs)
+        dao_klass, query, page, page_size = cls._make_query('article', None, q, page, page_size, sort, search_subs, sort_subs)
 
         # execute the query against the articles
-        res = models.Article.query(q=query.query(), consistent_order=False)
+        res = dao_klass.query(q=query.as_dict(), consistent_order=False)
```

Comment:
I think this should actually look more like this, to preserve the encapsulation provided by the queryService:

```
query = self._make_query(....)
res = articleService.search("api_query", "[type]", query, account)
```